### PR TITLE
linting test on GH actions.

### PR DESCRIPTION
**Pull request by nicolas-rabault**
Setting up a clean code linter with a fixed format on projects and automate it.
Actions need to raise an error in case of failure and display errors as diff.
The lining should be callable in the command line locally.

### CODE REVIEW

File: Projects/NUCLEO-F401RE/Gate/lib/Gate/cmd.c [hunk 1]
```diff
@@ -24,9 +24,7 @@ static void next_json(void)
 
 void check_json(uint16_t carac_nbr)
 {
-    if ((current_table > (JSON_BUF_NUM - 1)) ||
-        (current_table < 0) ||
-        (carac_nbr > (JSON_BUFF_SIZE - 1)))
+    if ((current_table > (JSON_BUF_NUM - 1)) || (current_table < 0) || (carac_nbr > (JSON_BUFF_SIZE - 1)))
```
Comment (JeromeGalan) [inline Projects/NUCLEO-F401RE/Gate/lib/Gate/cmd.c:27]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: Projects/NUCLEO-F410RB/Gate/lib/Gate/cmd.c [hunk 2]
```diff
@@ -24,9 +24,7 @@ static void next_json(void)
 
 void check_json(uint16_t carac_nbr)
 {
-    if ((current_table > (JSON_BUF_NUM - 1)) ||
-        (current_table < 0) ||
-        (carac_nbr > (JSON_BUFF_SIZE - 1)))
+    if ((current_table > (JSON_BUF_NUM - 1)) || (current_table < 0) || (carac_nbr > (JSON_BUFF_SIZE - 1)))
```
Comment (JeromeGalan) [inline Projects/NUCLEO-F410RB/Gate/lib/Gate/cmd.c:27]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: Projects/NUCLEO-L432KC/Gate/lib/Gate/cmd.c [hunk 3]
```diff
@@ -24,9 +24,7 @@ static void next_json(void)
 
 void check_json(uint16_t carac_nbr)
 {
-    if ((current_table > (JSON_BUF_NUM - 1)) ||
-        (current_table < 0) ||
-        (carac_nbr > (JSON_BUFF_SIZE - 1)))
+    if ((current_table > (JSON_BUF_NUM - 1)) || (current_table < 0) || (carac_nbr > (JSON_BUFF_SIZE - 1)))
```
Comment (JeromeGalan) [inline Projects/NUCLEO-L432KC/Gate/lib/Gate/cmd.c:27]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: Projects/STM32F4-discovery/Gate/lib/Gate/cmd.c [hunk 4]
```diff
@@ -24,9 +24,7 @@ static void next_json(void)
 
 void check_json(uint16_t carac_nbr)
 {
-    if ((current_table > (JSON_BUF_NUM - 1)) ||
-        (current_table < 0) ||
-        (carac_nbr > (JSON_BUFF_SIZE - 1)))
+    if ((current_table > (JSON_BUF_NUM - 1)) || (current_table < 0) || (carac_nbr > (JSON_BUFF_SIZE - 1)))
```
Comment (JeromeGalan) [inline Projects/STM32F4-discovery/Gate/lib/Gate/cmd.c:27]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: Projects/l0/Controller_motor/lib/Controller_motor/controller_motor.c [hunk 5]
```diff
@@ -395,14 +394,12 @@ static void ControllerMotor_MsgHandler(container_t *container, msg_t *msg)
         if (msg->header.size == sizeof(asserv_pid_t))
         {
             // fill the message infos
-            if ((motor.mode.mode_angular_position || motor.mode.mode_linear_position) &&
-                !(motor.mode.mode_angular_speed || motor.mode.mode_linear_speed))
+            if ((motor.mode.mode_angular_position || motor.mode.mode_linear_position) && !(motor.mode.mode_angular_speed || motor.mode.mode_linear_speed))
             {
```
Comment (JeromeGalan) [inline Projects/l0/Controller_motor/lib/Controller_motor/controller_motor.c:398]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: Projects/l0/Controller_motor/lib/Controller_motor/controller_motor.c [hunk 6]
```diff
@@ -395,14 +394,12 @@ static void ControllerMotor_MsgHandler(container_t *container, msg_t *msg)
...
                 // only position control is enable, we can save PID for positioning
                 memcpy(&position, msg->data, msg->header.size);
             }
-            if ((motor.mode.mode_angular_speed || motor.mode.mode_linear_speed) &&
-                !(motor.mode.mode_angular_position || motor.mode.mode_linear_position))
+            if ((motor.mode.mode_angular_speed || motor.mode.mode_linear_speed) && !(motor.mode.mode_angular_position || motor.mode.mode_linear_position))
```
Comment (JeromeGalan) [inline Projects/l0/Controller_motor/lib/Controller_motor/controller_motor.c:402]: Perhaps we should put OR and AND operators at line beginning.  So, after clang format processing, the conditions should be on different lines.

File: Projects/l0/Controller_motor/lib/Controller_motor/controller_motor.c [hunk 7]
```diff
@@ -629,8 +626,7 @@ void HAL_SYSTICK_Motor_Callback(void)
         last_systick = HAL_GetTick();
     }
     // ****** Linear interpolation *********
-    if ((motor.mode.mode_angular_position || motor.mode.mode_linear_position) &&
-        (motor.mode.mode_angular_speed || motor.mode.mode_linear_speed))
+    if ((motor.mode.mode_angular_position || motor.mode.mode_linear_position) && (motor.mode.mode_angular_speed || motor.mode.mode_linear_speed))
```
Comment (JeromeGalan) [inline Projects/l0/Controller_motor/lib/Controller_motor/controller_motor.c:629]: Perhaps we should put OR and AND operators at line beginning.  So, after clang format processing, the conditions should be on different lines.

File: Projects/NUCLEO-F401RE/Gate/lib/Gate/gate.c [hunk 8]
```diff
@@ -220,7 +224,8 @@ static int serial_write(char *data, int len)
 {
     for (unsigned short i = 0; i < len; i++)
     {
-        while (!LL_USART_IsActiveFlag_TXE(USART2));
+        while (!LL_USART_IsActiveFlag_TXE(USART2))
+            ;
```
Comment (JeromeGalan) [inline Projects/NUCLEO-F401RE/Gate/lib/Gate/gate.c:228]: Well, it's not very pretty but it seems there are no solutions...

File: Projects/l0/Imu/lib/Imu/mpu_configuration.c [hunk 9]
```diff
@@ -573,9 +573,7 @@ void mpu_setup(void)
 	     *
 	     * DMP sensor fusion works only with gyro at +-2000dps and accel +-2G
 	     */
-    hal.dmp_features = DMP_FEATURE_6X_LP_QUAT | DMP_FEATURE_TAP |
-                       DMP_FEATURE_ANDROID_ORIENT | DMP_FEATURE_SEND_RAW_ACCEL | DMP_FEATURE_SEND_CAL_GYRO |
-                       DMP_FEATURE_GYRO_CAL;
+    hal.dmp_features = DMP_FEATURE_6X_LP_QUAT | DMP_FEATURE_TAP | DMP_FEATURE_ANDROID_ORIENT | DMP_FEATURE_SEND_RAW_ACCEL | DMP_FEATURE_SEND_CAL_GYRO | DMP_FEATURE_GYRO_CAL;
```
Comment (JeromeGalan) [inline Projects/l0/Imu/lib/Imu/mpu_configuration.c:576]: In this case, do we need a logical OR (operator | )  or a or (operator || ) ?

Perhaps we should put OR operators at line beginning.  So, after clang format processing, the conditions should be on different lines.

File: Projects/NUCLEO-F401RE/Gate/lib/Gate/gate.c (hunk 8 above)
Comment (nicolas-rabault) [reply Projects/NUCLEO-F401RE/Gate/lib/Gate/gate.c:228]: yes, using clang-format we can't do it differently...

**Review by JeromeGalan — APPROVED** [review]

**Pull request merged**